libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -196,47 +219,51 @@ export class GRPCService implements Service {
     }
   }
 
-  private handleComplete() {
+  private handleClose() {
     this.logger?.info(`${FlagdProvider.name}: streaming connection closed gracefully`);
     this._cache?.clear();
     this._streamAlive = false;
   }
 
-  private objectParser = (struct: Struct) => {
+  private objectParser = (struct: unknown): unknown => {
     if (struct) {
-      return Struct.toJson(struct);
+      return struct;
     }
-    return {}
+    return {};
   };
 
-  private booleanParser = (value: boolean) => {
+  private booleanParser = (value: boolean): boolean => {
     if (value) {
       return value;
     }
-    return false
+    return false;
   };
 
-  private stringParser = (value: string) => {
+  private stringParser = (value: string): string => {
     if (value) {
       return value;
     }
-    return ''
+    return '';
   };
 
-  private numberParser = (value: number) => {
+  private numberParser = (value: number): number => {
     if (value) {
       return value;
     }
-    return 0
+    return 0;
   };
 
-  private async resolve<T extends FlagValue, Rq extends AnyRequest, Rs extends AnyResponse>(
-    resolver: (request: AnyRequest) => UnaryCall<Rq, Rs>,
+  private async resolve<T extends FlagValue>(
+    promise: (
+      request: AnyRequest,
+      callback: (error: ServiceError | null, response: AnyResponse) => void,
+    ) => ClientUnaryCall,
     flagKey: string,
     context: EvaluationContext,
     logger: Logger,
-    parser?: (value: any) => any
+    parser?: (value: any) => any,
```

Comment:
Can we avoid using `any` here?

Reply:
Probably.

Reply:
I've done this, though a bit of casting was needed, I think it improves type-safety. It only goes so far though b/c the underlying generated gRPC code returns an `any`.